corehq/messaging/scheduling/forms.py

```diff
@@ -2219,13 +2232,26 @@ def clean_user_data_property_name(self):
         return value
 
     def clean_user_data_property_value(self):
-        if self.cleaned_data.get('use_user_data_filter') != self.YES:
+        use_user_data_filter = self.cleaned_data.get('use_user_data_filter')
+        if use_user_data_filter == self.NO:
             return None
 
         value = self.cleaned_data.get('user_data_property_value')
         if not value:
             raise ValidationError(_("This field is required."))
 
+        if use_user_data_filter == self.JSON:
+            err = _("Invalid JSON value. Expected a list of strings.")
+            try:
+                value = json.loads(value)
+            except Exception:
+                raise ValidationError(err)
+            if (not isinstance(value, list) or
+                    any(not isinstance(v, six.text_type) for v in value)):
+                raise ValidationError(err)
```

Comment:
could we also validate that it has at least one element in the list?

Reply:
593ff4d352cc1e3c8425d369dde0c860daa397dc